### PR TITLE
let NibWriter accept file names

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request allows NibWriter in Bio.SeqIO.NibIO to accept file names in addition to file handles.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/SeqIO/NibIO.py [hunk 1]
```diff
@@ -138,15 +144,8 @@ def __init__(self, handle):
             raise RuntimeError("unexpected system byte order %s" % byteorder)
         handle.write(bytes.fromhex(signature))
 
-    def write_file(self, records):
+    def write_record(self, record):
         """Use this to write an entire file containing the given record."""
```
Comment (peterjc) [inline Bio/SeqIO/NibIO.py]: The comment seems a little confusing given you are using ``.write_record()`` now.

File: Bio/SeqIO/NibIO.py [hunk 2]
```diff
@@ -160,6 +159,14 @@ def write_file(self, records):
...
         indices = nucleotides.translate(table)
         handle.write(bytes.fromhex(indices))
+
+    def write_file(self, records):
+        """Use this to write an entire file containing the given records."""
+        count = super().write_file(records)
+        if count == 0:
+            raise ValueError("Must have one sequence")
+        if count > 1:
+            raise ValueError("More than one sequence found")
```
Comment (peterjc) [inline Bio/SeqIO/NibIO.py]: This seems raising an error after the fact - you might have just written a 1000 records giving a malformed NIB file. Surely better to fail immediately? A try/except with ``next()`` to confirm just one record seems better here.

File: Tests/test_SeqIO_write.py [hunk 3]
```diff
@@ -76,7 +76,7 @@
 
 
 class WriterTests(unittest.TestCase):
-    """Cunning unit test where methods are added at run time."""
+    """Cunning unit test where methods are added at run time."""  # TODO - Let's not be cunning
```
Comment (peterjc) [inline Tests/test_SeqIO_write.py]: Why? The only obvious alternative is a *lot* of boilerplate code defining test methods for example example file.

File: Bio/SeqIO/NibIO.py (hunk 2 above)
Comment (mdehoon) [reply Bio/SeqIO/NibIO.py]: Yes. But the best way to solve this is in the base class, which has its own problems. We'll get to that, don't worry.

File: Tests/test_SeqIO_write.py (hunk 3 above)
Comment (mdehoon) [reply Tests/test_SeqIO_write.py]: There must be a better way than what we have now. Currently when a test fails, it is unclear which test failed.

File: Bio/SeqIO/NibIO.py (hunk 1 above)
Comment (mdehoon) [reply Bio/SeqIO/NibIO.py]: OK

File: Bio/SeqIO/NibIO.py (hunk 2 above)
Comment (peterjc) [reply Bio/SeqIO/NibIO.py]: The main base class assumes multiple records, a subclass for single record formats would be a good general solution?

Comment (mdehoon) [reply Bio/SeqIO/NibIO.py]: Possibly. I am looking for a solution where only a single class holds the `handle` as an attribute.

Comment (peterjc) [reply Bio/SeqIO/NibIO.py]: That sounds compatible.

**Comment by mdehoon** [comment]
Can we move ahead with this one? Then we can have another look at the base class.

**Comment by peterjc** [comment]
OK then - if you're planning to revisit the NIB writer soon.

**Pull request merged**